src/compiler/checker.ts

```diff
@@ -12352,10 +12352,22 @@ namespace ts {
                 return getSymbolOfNode(node.parent);
             }
 
-            if (node.kind === SyntaxKind.Identifier && isInRightSideOfImportOrExportAssignment(<Identifier>node)) {
-                return node.parent.kind === SyntaxKind.ExportAssignment
-                    ? getSymbolOfEntityNameOrPropertyAccessExpression(<Identifier>node)
-                    : getSymbolOfPartOfRightHandSideOfImportEquals(<Identifier>node);
+            if (node.kind === SyntaxKind.Identifier) {
+                if (isInRightSideOfImportOrExportAssignment(<Identifier>node)) {
+                    return node.parent.kind === SyntaxKind.ExportAssignment
+                        ? getSymbolOfEntityNameOrPropertyAccessExpression(<Identifier>node)
+                        : getSymbolOfPartOfRightHandSideOfImportEquals(<Identifier>node);
+                }
+                else if (node.parent.kind === SyntaxKind.BindingElement &&
+                        node.parent.parent.kind === SyntaxKind.ObjectBindingPattern &&
+                        node === (<BindingElement>node.parent).propertyName) {
+                    let typeOfPattern = getTypeAtLocation(node.parent.parent);

```

Comment:
For a parameter with a binding pattern with no default initializer and no type annotation, this will return undefined, as you've noticed in the next line. This is because getTypeAtLocation ultimately calls getTypeForVariableLikeDeclaration, which doesn't handle the parameter case. If you were instead to hit getWidenedTypeForVariableLikeDeclaration, you would get to the call to getTypeFromBindingPattern, which handles the parameter case.